File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
  // const table = participant.reduce(
  //   (acc, cur) => ({ ...acc, [cur]: (acc[cur] || 0) + 1 }),
  //   {}
  // );
    let person={}
    for (let i of participant){
        if(typeof person[i]==="undefined"){
             person[i] = 1
        }else{ //동명이인 있을 수 있어
            person[i]+=1
        }
    }
    table = person

  completion.forEach((p) => {
    table[p] -= 1;
  });

  for ([name, isRemain] of Object.entries(table)) {
    if (isRemain) return name;
  }
}

File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participants, completion):
    marthoners = Marthoners()
    for participant in participants:
        marthoners.add(participant)
    for participant in completion:
        marthoners.remove(participant)
    return marthoners.remain

class Marthoners:
    def __init__(self):
        self._dict = {}
    def add(self,name):
        if name in self._dict:
            self._dict[name] += 1
        else:
            self._dict[name] = 1
    def remove(self, name):
        if self._dict[name] == 1:
            del self._dict[name]
        else:
            self._dict[name]-=1
    @property
    def remain(self):
        return list(self._dict.keys())[0]